src/main/java/org/esbtools/gateway/resync/ResyncServiceRepository.java

```diff
@@ -0,0 +1,28 @@
+package org.esbtools.gateway.resync;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.Set;
+
+@Repository
+public class ResyncServiceRepository {
+
+    private Set<ResyncService> resyncServices;
+
+    @Autowired
+    public ResyncServiceRepository(Set<ResyncService> resyncServices) {
+        this.resyncServices = resyncServices;
+    }
+
+    public ResyncService getBySystem(String system) {
+        ResyncService requestedService = null;
+        for (ResyncService resyncService : resyncServices) {
+            if (system.equals(resyncService.getSystemName())) {
+                requestedService = resyncService;
+                break;
+            }
+        }
+        return requestedService;
```

Comment:
IMHO better to throw/catch exception than return null / check for nulls... I generally try to completely eliminate nulls from programs so you never get a value of a type that isn't actually a thing of that type. Makes programs easier to reason about IMHO